chore: Unify some bb presets. Better devops to support native bb.js.

charlielye opened the pull request:
TLDR;
If you `./barretenberg/cpp/bootstrap.sh` it builds all the things.
If you `AVM=0 ./barretenberg/cpp/bootstrap.sh` it won't build vm2 or new bb-avm binary.

Prior to this we had the awkward need to rebuild *everything* twice just to get build of bb without the avm. Similar, but less impactful situation with the NAPI module. This is particularly bad now that we want to bundle "bb lite" (no avm) in bb.js, but still need "bb chonk" (with avm) in the container.

* We enable PIC by default as modern CPU's are basically 0 overhead.
* We completely remove the DISABLE_AZTEC_VM setting and #ifdefs.
* We delete a bunch of CMakePreset's.
* Instead we move some vm2 dsl code into vm2 module, and provide a stub module vm2_stub.
* We now have two build targets for bb, `bb` and `bb-avm`. The former links to the stub, the later to vm2 proper.
* As PIC is now enabled, we remove seperate NAPI presets and instead have target nodejs_module.
* We diverge bootstrap build if semver implies release. In which case we do additional cross compiles.
* bb.js similarly, will install all `bb` and napi module cross compiles into it's `build` folder if bootstrapping a release. *acutally NOT doing the copy in this PR, so I can land the "bb.js now defaults to native" in follow up.*

Still to do:
* Actually enable native bb.js
* Convert TS code that currently directly calls binary, to use bb.js api instead.

ludamad commented on barretenberg/ts/README.md:
````diff
@@ -85,7 +85,7 @@ You can enable these headers for specific pages that perform proof generation, b
 ```
-NO_STRIP=1 ./script/build_wasm.sh
+BUILD_CPP=1 NO_STRIP=1 ./script/copy_wasm.sh
 ```
````
I believe NO_STRIP is not picked up because cmake handles this

ludamad reviewed (approved):
Loving it

charlielye committed:
chore: Unify some bb presets. Better devops to support native bb.js.

TLDR;
If you `./barretenberg/cpp/bootstrap.sh` it builds all the things.
If you `AVM=0 ./barretenberg/cpp/bootstrap.sh` it won't build vm2 or new bb-avm binary.

Prior to this we had the awkward need to rebuild *everything* twice just to get build of bb without the avm. Similar, but less impactful situation with the NAPI module. This is particularly bad now that we want to bundle "bb lite" (no avm) in bb.js, but still need "bb chonk" (with avm) in the container.

* We enable PIC by default as modern CPU's are basically 0 overhead.
* We completely remove the DISABLE_AZTEC_VM setting and #ifdefs.
* We delete a bunch of CMakePreset's.
* Instead we move some vm2 dsl code into vm2 module, and provide a stub module vm2_stub.
* We now have two build targets for bb, `bb` and `bb-avm`. The former links to the stub, the later to vm2 proper.
* As PIC is now enabled, we remove seperate NAPI presets and instead have target nodejs_module.
* We diverge bootstrap build if semver implies release. In which case we do additional cross compiles.
* bb.js similarly, will install all `bb` and napi module cross compiles into it's `build` folder if bootstrapping a release. *acutally NOT doing the copy in this PR, so I can land the "bb.js now defaults to native" in follow up.*

Still to do:
* Actually enable native bb.js
* Convert TS code that currently directly calls binary, to use bb.js api instead.

fcarreiro commented on barretenberg/cpp/CMakePresets.json:
```diff
@@ -63,6 +47,21 @@
     },
+    {
+      "name": "clang20-no-avm",
+      "displayName": "Build with Clang-20 plus AVM",
+      "description": "Build with globally installed Clang-20 plus AVM",
```
the names says "no avm" but the description says "with avm"?
